.github/workflows/sycl-linux-run-tests.yml

```diff
@@ -288,23 +288,51 @@ jobs:
       if: inputs.tests_selector == 'cts'
       env:
         CMAKE_EXTRA_ARGS: ${{ inputs.extra_cmake_args }}
+      # Continue execution if some category fails to build.
+      shell: bash {0}
+      # Compile each binary individually; if a failure occurs during the
+      # 'test_all' build, the entire build will be affected.
       run: |
         cmake -GNinja -B./build-cts -S./khronos_sycl_cts -DCMAKE_CXX_COMPILER=$(which clang++) \
         -DSYCL_IMPLEMENTATION=DPCPP \
         -DSYCL_CTS_EXCLUDE_TEST_CATEGORIES="$PWD/devops/cts_exclude_filter" \
         -DSYCL_CTS_ENABLE_OPENCL_INTEROP_TESTS=OFF \
         -DSYCL_CTS_MEASURE_BUILD_TIMES=ON \
-        -DDPCPP_INSTALL_DIR="$$(dirname (which clang++))/.." \
-        $CMAKE_EXTRA_ARGS
-        ninja -C build-cts
+        -DDPCPP_INSTALL_DIR="$(dirname $(which clang++))/.." \
+        $CMAKE_EXTRA_ARGS | tee "cmake_output.txt"
+
+        # Get names of all added test-categories:
+        categories=$(grep -o '^-- Adding test: test_[a-zA-Z0-9_]*' "./cmake_output.txt" | sed 's/^-- Adding test: //')
+
+        number_of_test_categories=$(echo $categories | wc -w)
+        iteration=1
+        failed_to_build=""
+
+        for i in $categories; do
+          echo "::group::Building $i ($iteration of $number_of_test_categories)"
+          ninja -C build-cts $i
+          [ $? -ne 0 ] && failed_to_build+=$i'\n'
+          echo "::endgroup::"
+          ((iteration++))
+        done
+
+        if [ -n "$failed_to_build" ]; then
+          echo -e "Failed to build:\n$failed_to_build"
+          echo -e "Failed to build:\n$failed_to_build" >> $GITHUB_STEP_SUMMARY
+          exit 1
+        fi
+
+        exit 0
 
     - name: SYCL CTS List devices
-      if: inputs.tests_selector == 'cts'
+      # Proceed with execution even if the 'build' step did not succeed.
+      if: inputs.tests_selector == 'cts' && (success() || failure())
       run: |
-        ./build-cts/bin/test_all --list-devices
+        ./build-cts/bin/test_accessor --list-devices
```

Comment:
That won't work if that particular test has failed. I'd suggest to list devices/run each tests immediately after that test's build. WDYT?

Reply:
I'd probably prefer not to mix the results of "build" and "run".

Updated this string - [79b10c4](https://github.com/intel/llvm/pull/13421/commits/79b10c48aab7151c94663acfa215dbec167d1cb6)